new things for nuxt

FunctionalMetatable opened the pull request:
Stuff for Nuxt.

Includes:
Login System
Auth Middleware
Header/Footer pages
Tutorial Component
Renderer Component <= TODO: Code Syntax Highlighting Library in vue & project embed. cc @CubeyTheCube
Markdown Tutorial
Prettier Workflow
Contributors Page <= TODO: Proper animation

...and more. Not sure if this is gonna get merged anyways but ok

You can view it at https://flagclicked-nuxtification.9gr.vercel.app/

TheColaber commented on nuxt.config.js:
```diff
@@ -4,37 +4,41 @@ export default {
    */
   head: {
-    title: "Flag Clicked",
+    titleTemplate: (chunk) => {
+      return chunk ? `${chunk} - When Flag Clicked` : "When Flag Clicked";
+    },
```
I never knew about this but it looks great... I would like it to say something `How to create a platformer on When Flag Clicked` if it was on a tutorial. That would be pretty cool. Are there also any for the meta stuff? It's fine if this is not in this PR we could open another issue for it.

FunctionalMetatable replied:
i've been learning nuxt from ocular's source code

TheColaber commented on nuxt.config.js:
```diff
@@ -51,5 +55,9 @@ export default {
-  plugins: []
+  plugins: [],
+
+  env: {
+    backendURL: "https://api.whenflagclicked.org",
+  },
```
<3 - btw this url has not been created yet but I'll tell ryker to create it

TheColaber commented on pages/tutorials/_id/index.vue:
```diff
@@ -0,0 +1,36 @@
+<script>
+export default {
+  head() {
+    title: "Loading Tutorial"
```
TODO - I'll explain later

FunctionalMetatable replied:
heh i forgot return

TheColaber reviewed (commented):
Epic codes once again... I'll reveiw some more later and contribute a bit more once this and your other PR gets merged 👍

TheColaber commented:
NOTE: (you may already know this but) A middleware should be in the frontend nuxt that gets the token in the user's cookies and verifies that the user's token is valid. If not we should remove the cookie and therefore they will be logged out.

TheColaber commented:
Hey btw how did you get this on vercel??? I've always tried to but it always fails lol

TheColaber commented:
I would like an action where the bot creates a vercel link on every PR... maybe you or I can create an issue on this to discuss stuff.

FunctionalMetatable commented:
> Hey btw how did you get this on vercel??? I've always tried to but it always fails lol

header kinda broken.
for me, i have a hobby account so i can deploy it. i think you cant as this is under an organization. i'll see what i can do

TheColaber commented on layouts/default.vue:
```diff
@@ -1,16 +1,11 @@
 <template>
   <div class="container">
-    <Navbar />
```
Navbar was removed

FunctionalMetatable replied:
will fix soon

TheColaber commented on pages/index.vue:
```diff
@@ -1,10 +1,13 @@
-    </div>
-  </center>
+  <div class="container">
+    <Navbar />
```
Why only in index.vue? Why not on every page?

FunctionalMetatable replied:
done

TheColaber commented:
@FunctionalMetatable I think we got vercel to work :tada: Ready to merge? Then I will move backend here...

TheColaber commented:
(I do see a couple of issues with the css but I'll fix those later :+1:

FunctionalMetatable commented:
> (I do see a couple of issues with the css but I'll fix those later 👍

the navbar's position set to absolute just breaks the page

FunctionalMetatable commented:
done middleware

FunctionalMetatable commented:
@TheColaber could you authorize the deployments?

FunctionalMetatable commented:
Deploying perfectly for me.
@TheColaber please could you review?

CubeyTheCube commented:
when this is deployed will I see this on flagclicked.vercel.app?

FunctionalMetatable commented:
> when this is deployed will I see this on flagclicked.vercel.app?

https://flagclicked-nuxtification.9gr.vercel.app/

FunctionalMetatable commented:
not sure what I should add to this pr

TheColaber commented:
@FunctionalMetatable sorry for the wait, I'm trying to figure out what we should do with the auth and stuff.

TheColaber commented:
@FunctionalMetatable Take a look at https://github.com/nuxt-community/markdownit-module. Use it instead of the other markdown library that's currently in use. Our goal is to move away from our libraries that aren't nuxty and move to the nuxty ones.

FunctionalMetatable commented:
> @FunctionalMetatable Take a look at https://github.com/nuxt-community/markdownit-module. Use it instead of the other markdown library that's currently in use. Our goal is to move away from our libraries that aren't nuxty and move to the nuxty ones.

Marked works fine. It's an npm module, so it can be bundled client-side as well. With marked, we can render scratchblocks, add our own flavored markdown, etc. With the _nuxty_ ones we cannot actually implement our own flavored markdown with

markverb1 commented:
> > @FunctionalMetatable Take a look at https://github.com/nuxt-community/markdownit-module. Use it instead of the other markdown library that's currently in use. Our goal is to move away from our libraries that aren't nuxty and move to the nuxty ones.
> 
> Marked works fine. It's an npm module, so it can be bundled client-side as well. With marked, we can render scratchblocks, add our own flavored markdown, etc. ~With the _nuxty_ ones we cannot actually implement our own flavored markdown with~
> Edit: I'll check the markdown-it plugins thing

i think you can add plugins or whatever you call them
###### https://stackoverflow.com/questions/59680780/creating-a-custom-markdown-it-plugin-for-nuxt-application-results-in-typeerror-w

FunctionalMetatable commented:
I replaced `marked` with `markdown-it` and added Syntax highlighting. Could this get merged/reviewed? (I personally **don't** want to be doing the whole nuxtification)

TheColaber commented:
Thanks for your hard work, I'll merge this and work on a couple things later.
